reframe/core/pipeline.py

```diff
@@ -1095,10 +1095,13 @@ def cleanup(self, remove_files=False, unload_env=True):
             self._current_partition.local_env.unload()
 
     def __str__(self):
-        return ('%s (%s)\n'
-                '        tags: [%s], maintainers: [%s]' %
-                (self.name, self.descr,
-                 ', '.join(self.tags), ', '.join(self.maintainers)))
+        return ('%s (found in %s)\n'
+                '        descr: %s\n'
+                '        tags: {%s}, maintainers: [%s]' %
```

Comment:
Here we could directly print the set and the maintainers, without needing to have `{}` or `[]`.